book/HAS/hybrid-application-service-design.md

```diff
@@ -0,0 +1,88 @@
+
+## Application CR Update
+
+Application model update can be done by updating an existing Application CR directly. Application CR update is handled by the Application Controller. Only the mutable fields listed in the required fields in the Application CR can be updated.
+
+When an Application CR is being updated, the Application Controller will:
+1. Check to make sure all changed fields are mutable. Any changes on the immutable fields will result in an error in the Application CR Update.
+2. Update the application devfile (as specified in the spec.appModelRepository.url if applicable)
+
+## Component CR Update
+
+Component model update can be done by updating an existing Component CR directly. Component CR update is handled by the Component Controller. Only the mutable fields listed in the required fields in the Application CR can be updated.
+
+When an Component CR is being updated, the Component Controller will:
+1. Check to make sure all changed fields are mutable. Any changes on the immutable fields will result in an error in the Component CR Update.
+2. Update the component devfile (as specified in the status.data if applicable)
+3. For changes that require GitOps repo changes, call the HAS internal Push function to update the GitOps repository to reflect the model change for fields.
+
+## Usage of Devfile in Applications and Components
+
+The git source repo may or may not have the devfile physically stored within the repo. In case the repo has a devfile exists, it may not be the latest version. Future plan is to provide a way for creating PRs to put the latest devfile in the user’s source repo, the user may not accept the PR. Therefore, the devfile within the Application CRs and Component CRs will always be the latest and any action that needs to use the devfile stored in the Component CR to make sure it is using the latest version of the devfile, e.g. build pipeline and GitOps resource generation.
+
+## Build Service Integration
+
+For the Build Service integration, The HAS Component controller will create a webhook url during the Component creation process. A TriggerTemplate and an EventListener will be created as per the documentation AppStudio Build Service Architecture
+
+The webhook URL will only be created with a user provided Git repository. If the user chooses to import from a Devfile Registry sample, the user would need to clone the sample and import it via HAC to enable webhooks for the Git repository.
+
+The webhook URL would be displayed on HAC and the user is expected to manage it manually via Git.
+
+## GitOps Service Integration
+
+For the GitOps service integration, the HAS controller will create the GitOps repository when Application CRs are created. And when Component CRs are created/updated, will generate and push the corresponding GitOps resources for that component to the repository (as outlined above for Application and Component create/update).
```

Comment:
Edit second sentence:

`When Component CRs are created/updated, corresponding GitOps resources will be generated in the repository (as outlined above for Application and Component create/update).`

We don't really describe the generation of files under the Component creation scenario.  I'll add a comment in that section